UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs

```diff
@@ -310,6 +314,96 @@ public async Task<string> SearchReadMeFilePath(string batchId,string correlation
+                return true;
+            }
+            else
+            {
+                logger.LogError(EventIds.ErrorInCreatingZipFile.ToEventId(), "Error in creating V01X01.zip with _X-Correlation-ID:{correlationId}", correlationId);
+                return false;
+            }
+        }
+
+        public async Task<bool> UploadZipFileForExchangeSetToFileShareService(string batchId, string exchangeSetZipRootPath, string correlationId)
+        {
+            var accessToken = await authTokenProvider.GetManagedIdentityAuthAsync(fileShareServiceConfig.Value.ResourceId);
+            bool isUploadZipFile = false;
+            CustomFileInfo customFileInfo = fileSystemHelper.GetFileInfo(Path.Combine(exchangeSetZipRootPath, fileShareServiceConfig.Value.ExchangeSetFileName));
+
+            bool isZipFileCreated = await CreateExchangeSetFile(batchId, correlationId, accessToken, customFileInfo);
+            if (isZipFileCreated)
+            {
+                bool isWriteBlock = await UploadAndWriteBlock(batchId, correlationId, accessToken, customFileInfo);
+                if (isWriteBlock)
+                {
+                    string batchStatus = await CommitAndGetBatchStatus(batchId, correlationId, accessToken, customFileInfo);
+                    if (batchStatus == BatchStatus.Committed.ToString())
+                    {
+                        isUploadZipFile = true;
+                    }
+                }
+            }
+            return isUploadZipFile;
+        }
+
+        public async Task<bool> CreateExchangeSetFile(string batchId, string correlationId, string accessToken, CustomFileInfo customFileInfo)
+        {
+            FileCreateMetaData fileCreateMetaData = new FileCreateMetaData()
+            {
+                AccessToken = accessToken,
+                BatchId = batchId,
+                FileName = customFileInfo.Name,
+                Length = customFileInfo.Length
+            };
+            return await CreateFile(fileCreateMetaData, accessToken, correlationId);
+        }
+
+        public async Task<string> CommitAndGetBatchStatus(string batchId, string correlationId, string accessToken, CustomFileInfo customFileInfo)
+        {
+            BatchCommitMetaData batchCommitMetaData = new BatchCommitMetaData()
+            {
+                BatchId = batchId,
+                AccessToken = accessToken,
+                FullFileName = customFileInfo.FullName
+            };
+            bool isBatchCommitted = await UploadCommitBatch(batchCommitMetaData, correlationId);
+            string batchStatus = BatchStatus.CommitInProgress.ToString();
+            if (isBatchCommitted)
+            {
+                var batchStatusMetaData = new BatchStatusMetaData()
+                {
+                    AccessToken = accessToken,
+                    BatchId = batchId
+                };
+                if (batchStatus != BatchStatus.Committed.ToString())
```

Comment:
Does this if statement add any value? (I'm assuming it will always be true)

Reply:
Removed If statement.